2023q3: dpaa2: Status update (#259)
freebsd/freebsd-doc: website/content/en/status/report-2023-07-2023-09/dpaa2.adoc

=== NXP DPAA2 support

> Links +
> link:https://cgit.freebsd.org/src/tree/sys/dev/dpaa2[DPAA2 in the FreeBSD source tree] +
> link:https://github.com/mcusim/freebsd-src[DPAA2 on Github]

Reviewer: ```suggestion
Links: +
link:https://cgit.freebsd.org/src/tree/sys/dev/dpaa2[DPAA2 in the FreeBSD source tree] URL: link:https://cgit.freebsd.org/src/tree/sys/dev/dpaa2[] +
link:https://github.com/mcusim/freebsd-src[DPAA2 on Github] URL: link:https://cgit.freebsd.org/src/tree/sys/dev/dpaa2[]
```

Whilst the repetitions, prefixes etc. do appear peculiar, some peculiarities are currently required for the eventual plain text edition of the report to include visible or usable links. See, for example, last quarter's:

* https://lists.freebsd.org/archives/freebsd-announce/2023-July/000077.html

=== NXP DPAA2 support

Links: +

> link:https://cgit.freebsd.org/src/tree/sys/dev/dpaa2[DPAA2 in the FreeBSD source tree] +
> URL: link:https://cgit.freebsd.org/src/tree/sys/dev/dpaa2[] +
> link:https://github.com/mcusim/freebsd-src[DPAA2 on Github] +
> URL: link:https://github.com/mcusim/freebsd-src[]

Reviewer 2: ```suggestion
link:https://cgit.freebsd.org/src/tree/sys/dev/dpaa2[DPAA2 in the FreeBSD source tree] URL: link:https://cgit.freebsd.org/src/tree/sys/dev/dpaa2[] +
link:https://github.com/mcusim/freebsd-src[DPAA2 on Github] URL: link:https://github.com/mcusim/freebsd-src[]
```

==== Changes from the previous report

* Isolation between DPAA2 channels link:https://cgit.freebsd.org/src/commit/?id=58983e4b0253ad38a3e1ef2166fedd3133fdb552[improved].
* Panic under heavy network load link:https://github.com/mcusim/freebsd-src/issues/19[fixed].
* FDT/ACPI MDIO support.
* NFS root mount link:https://github.com/mcusim/freebsd-src/issues/7[doesn't hang] on netboot over DPAA2 anymore.
* Drivers link:https://github.com/mcusim/freebsd-src/issues/2[started] to communicate with MC via their own command portals (DPMCP).
* link:https://github.com/mcusim/freebsd-src/issues?q=is%3Aissue+is%3Aclosed[List of all closed issues].

==== Work in Progress

Work on link:https://cgit.freebsd.org/src/commit/?id=2a9021898c4ee2154787da862c238cfeccd655df[dev/sff] started to support SFF/SFP modules in order to test DPAA2 drivers on links above 1 Gbit/s.

==== Plan

* Heavy network load tests (2.5 Gbit/s, 10 Gbit/s) and bottlenecks mitigation.
* Cached memory-backed software portals.
* Driver resources de-allocation to unload dpaa2.ko properly.
* Further parts (DPSW, DCE, etc.) supported by the hardware.

> Sponsor: Bare Enthusiasm :) +

Reviewer 2: I am sure that this is one of the most important sponsors, but I discourage listing it: I guess it holds for almost any project, it does not give any new real information, and there is also the risk that the importance of the real economical sponsor is underestimated.

=== NXP DPAA2 support

Links +
link:https://cgit.freebsd.org/src/tree/sys/dev/dpaa2[DPAA2 in the FreeBSD source tree] +
link:https://github.com/mcusim/freebsd-src[DPAA2 on Github]

Contact: Dmitry Salychev <dsl@FreeBSD.org> +
Contact: Bjoern A. Zeeb <bz@FreeBSD.org>

==== What is DPAA2?

> DPAA2 is a hardware-level networking architecture found in some NXP SoCs which contain hardware blocks including Management Complex (MC, a command interface to manipulate DPAA2 objects), Wire Rate I/O processor (WRIOP, packets distribution, queuing, drop decisions),
> Queues and Buffers Manager (QBMan, Rx/Tx queues control, Rx buffer pools) and the others. The Management Complex runs NXP-supplied firmware which provides DPAA2 objects as an abstraction layer over those blocks to simplify an access to the underlying hardware.

Reviewer: ```suggestion
DPAA2 is a hardware-level networking architecture found in some NXP SoCs which contain hardware blocks including Management Complex (MC, a command interface to manipulate DPAA2 objects), Wire Rate I/O processor (WRIOP, packets distribution, queuing, drop decisions), Queues and Buffers Manager (QBMan, Rx/Tx queues control, Rx buffer pools) and others.
The Management Complex runs NXP-supplied firmware which provides DPAA2 objects as an abstraction layer over those blocks to simplify access to the underlying hardware.
```

Reviewer: > … blocks including … Management Complex …, Queues and Buffers Manager (…) and others.

That's _including_, not _comprising_, so you can lose the two words at the tail of the sentence: 

> … blocks including … Management Complex …, and Queues and Buffers Manager (…).

==== Changes from the previous report

> * Isolation between DPAA2 channels link:https://cgit.freebsd.org/src/commit/?id=58983e4b0253ad38a3e1ef2166fedd3133fdb552[improved].
> 
> * Panic under heavy network load link:https://github.com/mcusim/freebsd-src/issues/19[fixed].
> 
> * FDT/ACPI MDIO support.
> 
> * NFS root mount link:https://github.com/mcusim/freebsd-src/issues/7[doesn't hang] on netboot over DPAA2 anymore.
> 
> * Drivers link:https://github.com/mcusim/freebsd-src/issues/2[started] to communicate with MC via their own command portals (DPMCP).
> 
> * link:https://github.com/mcusim/freebsd-src/issues?q=is%3Aissue+is%3Aclosed[List of all closed issues].

Reviewer: ```suggestion
* Isolation between DPAA2 channels is link:https://cgit.freebsd.org/src/commit/?id=58983e4b0253ad38a3e1ef2166fedd3133fdb552[improved].
* Panic under heavy network load is link:https://github.com/mcusim/freebsd-src/issues/19[fixed].
* FDT/ACPI MDIO support.
* NFS root mount link:https://github.com/mcusim/freebsd-src/issues/7[no longer hangs] on netboot over DPAA2.
* Drivers link:https://github.com/mcusim/freebsd-src/issues/2[started] to communicate with MC via their own command portals (DPMCP).
* link:https://github.com/mcusim/freebsd-src/issues?q=is%3Aissue+is%3Aclosed[List of all closed issues].
```

==== Work in Progress

> * Work on link:https://cgit.freebsd.org/src/commit/?id=2a9021898c4ee2154787da862c238cfeccd655df[dev/sff] started to support SFF/SFP modules in order to test DPAA2 drivers on the links above 1 Gbit/s

Reviewer: ```suggestion
Work on link:https://cgit.freebsd.org/src/commit/?id=2a9021898c4ee2154787da862c238cfeccd655df[dev/sff] started to support SFF/SFP modules in order to test DPAA2 drivers on links above 1 Gbit/s.
```

==== Plan

> * Heavy network load tests (2.5 Gbit/s, 10 Gbit/s) and bottlenecks mitigation.
> 
> * Cached memory-backed software portals.
> 
> * Driver resources de-allocation to unload dpaa2.ko properly.
> 
> * Further parts (DPSW, DCE, etc.) supported by the hardware.

Reviewer: ```suggestion
* Heavy network load tests (2.5 Gbit/s, 10 Gbit/s) and bottlenecks mitigation.
* Cached memory-backed software portals.
* Driver resources de-allocation to unload `dpaa2.ko` properly.
* Further parts (DPSW, DCE, etc.) supported by the hardware.
```
